HealthcareApp/src/Healthcare.Application/Appointments/Queries/GetRequestedAppointmentsQueryHandler.cs

```diff
@@ -0,0 +1,34 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Healthcare.Application.Appointments.Queries;
+
+public class
+    GetRequestedAppointmentsQueryHandler : IRequestHandler<GetRequestedAppointmentsQuery, List<Appointment>>
+{
+    private readonly IAppointmentRepository _appointmentRepository;
+    private readonly IUserManagerDecorator<ApplicationUser> _userManager;
+
+
+    public GetRequestedAppointmentsQueryHandler(IAppointmentRepository appointmentRepository,
+        IUserManagerDecorator<ApplicationUser> userManager)
+    {
+        _appointmentRepository = appointmentRepository;
+        _userManager = userManager;
+    }
+
+    public async Task<List<Appointment>> Handle(GetRequestedAppointmentsQuery request,
+        CancellationToken cancellationToken)
+    {
+        var users = await _userManager.GetAllUsersAsync();
+
+        var doctor = await users.FirstOrDefaultAsync(x => x.Id == request.DoctorId, cancellationToken);
+        if (doctor == null) throw new Exception("Doctor not found");
+
+
+        return await _appointmentRepository.GetRequestedAppointments(request.DoctorId);
```

Comment:
A single doctor can have millions of appointments when working for years. Use paging/filtering for such methods